vms/evm/destination_client.go

```diff
@@ -208,26 +280,76 @@ func (c *destinationClient) SendTx(
 			"Failed to sign transaction",
 			zap.Error(err),
 		)
-		return common.Hash{}, err
+		return nil, err
 	}
 
 	sendTxCtx, sendTxCtxCancel := context.WithTimeout(context.Background(), utils.DefaultRPCTimeout)
 	defer sendTxCtxCancel()
+
+	c.logger.Info(
+		"Sending transaction",
+		zap.String("txID", signedTx.Hash().String()),
+		zap.Uint64("nonce", c.currentNonce),
+		zap.Int("poolTxSlotsAvailable", cap(c.poolTxsSemaphore)-len(c.poolTxsSemaphore)),
+	)
+
+	// Acquire a semaphore to limit the number of transactions in the mempool
+	c.poolTxsSemaphore <- struct{}{}
```

Comment:
The same goes with this. If this function errors, we never release the semaphore.